lib/rubanok/dsl/matching.rb

```diff
@@ -64,16 +64,31 @@ def build_method_name
         end
       end
 
+      module InstanceMethods
+        def default_match_handler(rule, params, fail_when_no_matches)
+          fail_when_no_matches = Rubanok.fail_when_no_matches if fail_when_no_matches.nil?
+          return raw unless fail_when_no_matches
+
+          fail ::Rubanok::BadValueError, <<~MSG
```

Comment:
https://github.com/rubocop-hq/ruby-style-guide#prefer-raise-over-fail

Or at least be [consistent](https://github.com/palkan/rubanok/search?q=raise&unscoped_q=raise).